### ISSUE TITLE
search is unpredictable

### ISSUE THREAD

**Issue by dustymc** [ArctosDB/documentation-wiki#40]
Search is not returning expected results. I just spent way too long looking for something I wrote; nobody else would have found it.

<img width="338" alt="screen shot 2017-06-05 at 8 26 00 am" src="https://cloud.githubusercontent.com/assets/5720791/26790482/b5c537ca-49c8-11e7-9fb1-adbcb29e33c4.png">

"Locality" does not, for some reason, find http://handbook.arctosdb.org/how_to/How-to-understand-the-Arctos-Locality-Model.html

Also, why can't I down-arrow and enter to select a search result? Being forced to type and click is mildly annoying.

**Comment by tigerhawkvok** [comment]
I'll have to take around the library settings to see why it's not showing up in the first place. I could understand if it was ranked low, but it still should show up unless it is hitting some case that is preventing it from doing so.

With regards to the shortcut keys, I'll have to create some JavaScript events.

_Sent from my Huawei Nexus 6P using [FastHub](https://play.google.com/store/apps/details?id=com.fastaccess.github)_

**Comment by tigerhawkvok** [comment]
I forked and rewrote part of the search library.

It was previously going through the doc list roughly-but-not-necessarily-exactly by URL, pushing matches on to an array, and stopping when it hit the match limit (25).

I changed it to find a list of matches in several different match types, then build the final match resort by first populating from title, then url, then tag, then all other matches up to the limit.

This should change the erratic behaviour you saw and make it behave more "sanely":

![image](https://user-images.githubusercontent.com/165937/26864930-62738b42-4b0f-11e7-877f-11095f566f22.png)

I'll break the shortcut keys out into a new issue. If there continue to be search problems, let me know!
